bindings/python/gnucash_core.py

```diff
@@ -95,13 +95,21 @@ def __init__(self, book_uri=None, ignore_lock=False, is_new=False,
         ignore_lock is passed to qof_session_begin's argument of the
         same name and is used to break an existing lock on a dataset.
 
+        instance argument can be passed if new Session is used as a
+        wrapper for an existing session instance
 
 
         This function can raise a GnuCashBackendException. If it does,
         you don't need to cleanup and call end() and destroy(), that is handled
         for you, and the exception is raised.
         """
-        GnuCashCoreClass.__init__(self)
+        if instance:
+            GnuCashCoreClass.__init__(self, instance=instance)
+        else:
+            # even if instance is None GnuCashCoreClass.__init__ will still
+            # not create a new instance, only omitting the instance parameter
+            # works. Maybe change that?
```

Comment:
Yes, change that. Otherwise you risk passing `None` to the C API and it will crash.